src/main/java/net/fabricmc/loom/build/nesting/NestedDependencyProvider.java

```diff
@@ -103,17 +103,18 @@ public static NestedDependencyProvider createNestedDependencyProviderFromConfigu
 
 				visited.add(dependency.getGroup() + ":" + dependency.getName() + ":" + dependency.getVersion());
 
-				// TODO change this to allow just normal jar tasks, so a project can have a none loom sub project
 				Collection<Task> remapJarTasks = dependencyProject.getTasksByName("remapJar", false);
 				Collection<Task> jarTasks = dependencyProject.getTasksByName("jar", false);
 
 				for (Task task : remapJarTasks.isEmpty() ? jarTasks : remapJarTasks) {
-					if (task instanceof RemapJarTask remapJarTask) {
-						File file = remapJarTask.getArchiveFile().get().getAsFile();
-						fileList.add(new DependencyInfo<>(projectDependency, new ProjectDependencyMetaExtractor(), file));
-					} else if (task instanceof AbstractArchiveTask abstractArchiveTask) {
+					if (task instanceof AbstractArchiveTask abstractArchiveTask) {
 						File file = abstractArchiveTask.getArchiveFile().get().getAsFile();
-						fileList.add(new DependencyInfo<>(projectDependency, new ProjectDependencyMetaExtractor(), file));
+						fileList.add(new DependencyInfo<>(
+								projectDependency,
+								new ProjectDependencyMetaExtractor(),
+								file,
+								abstractArchiveTask.getArchiveClassifier().getOrNull()
+						));
```

Comment:
I don't think there's much of a point to keeping the `file` variable that helps with readability here if you are placing arguments on individual lines.